packages/modules/devices/solaredge/device.py

```diff
@@ -68,10 +71,30 @@ def add_component(self,
                 self.device_config.id, component_config, self.client))
             if component_type == "inverter" or component_type == "external_inverter":
                 self.inverter_counter += 1
-            self.synergy_units = int(self.client.read_holding_registers(
-                40129, modbus.ModbusDataType.UINT_16,
-                unit=component_config.configuration.modbus_id)) or 1
-            log.debug("Synergy Units: %s", self.synergy_units)
+            # try:
+            #     # ToDo: convert to String
+            #     manufacturer = self.client.read_holding_registers(40004, [modbus.ModbusDataType.UINT_32]*8,
+            #                                                       unit=component_config.configuration.modbus_id)
+            #     # ToDo: convert to String
+            #     model = self.client.read_holding_registers(40020, [modbus.ModbusDataType.UINT_32]*8,
+            #                                                unit=component_config.configuration.modbus_id)
+            #     # ToDo: convert to String
+            #     version = self.client.read_holding_registers(40044, [modbus.ModbusDataType.UINT_16]*8,
+            #                                                  unit=component_config.configuration.modbus_id)
+            #     serial_number = self.client.read_holding_registers(40052, [modbus.ModbusDataType.UINT_32]*8,
+            #                                                        unit=component_config.configuration.modbus_id)
+            #     log.debug("Version: " + str(version))
+            # except Exception as e:
+            #     log.exception("Fehler beim Auslesen der Modbus-Register: " + str(e))
+            #     pass
+            if self.client.read_holding_registers(40121, modbus.ModbusDataType.UINT_16,
+                                                  unit=component_config.configuration.modbus_id
+                                                  ) == synergy_unit_identifier:
+                log.debug("Synergy Units supported")
+                self.synergy_units = int(self.client.read_holding_registers(
+                    40129, modbus.ModbusDataType.UINT_16,
+                    unit=component_config.configuration.modbus_id)) or 1
```

Comment:
```suggestion
                self.synergy_units = int(self.client.read_holding_registers(
                    40129, modbus.ModbusDataType.UINT_16,
                    unit=component_config.configuration.modbus_id))
```
Die Variable wird schon beim Instanziieren auf 1 gesetzt.

Reply:
Ich habe den Code nur so übernommen. :-)
Habe auch kein System mit "Synergy Units" zum Testen, daher würde ich es jetzt vorerst dabei belassen.